### ISSUE TITLE
Add `string_t` module

### ISSUE THREAD

**Issue by constantinosgeorgiou** [constantinosgeorgiou/c-toolbox#55]
Make a string module. Adopt C++ `std::string` class to C. Create an interface as well as an implementation.

### Interface

Main structure: `string_t` which is a pointer to a struct `string_t`.

Should contain the following functions:

- string_create(): Allocates memory for new string.
- string_destroy(): De-allocates memory of string.
- string_length(): Return length of string, type `ssize_t`.
- string_concat(): Concatenate two string.
- string_assign(): Assign value to string.
- string_data(): Return c string. 

Basically take inspiration from here: [C++ Reference std::string](https://www.cplusplus.com/reference/string/string/)

Could be challenge of the week.
